Cannot use wrapped JSON Extractor for responses based on derive_from_request example

The code in the [custom-extractor-error example](https://github.com/tokio-rs/axum/blob/52a90390195e884bcc12ff5bd9fd805cac806447/examples/customize-extractor-error/src/derive_from_request.rs#L21) for deriving `FromRequest` for a custom wrapper Json extractor does not compile as soon as one removes the semicolon in the handler (i.e. actually returning the response that is created from `Json(dbg!(value))`):

```rust
pub async fn handler(Json(value): Json<Value>) -> impl IntoResponse {
    Json(dbg!(value))
}
```

```
error[E0277]: the trait bound `derive_from_request::Json<serde_json::Value>: IntoResponse` is not satisfied
  --> examples/customize-extractor-error/src/derive_from_request.rs:20:69
   |
20 |   pub async fn handler(Json(value): Json<Value>) -> impl IntoResponse {
   |  _____________________________________________________________________^
21 | |     Json(dbg!(value))
22 | | }
   | |_^ the trait `IntoResponse` is not implemented for `derive_from_request::Json<serde_json::Value>`
   |
   = help: the following other types implement trait `IntoResponse`:
             &'static [u8; N]
             &'static [u8]
             &'static str
             ()
             (R,)
             (Response<()>, R)
             (Response<()>, T1, R)
             (Response<()>, T1, T2, R)
           and 133 others

error[E0277]: the trait bound `derive_from_request::Json<serde_json::Value>: IntoResponse` is not satisfied
  --> examples/customize-extractor-error/src/derive_from_request.rs:20:51
   |
20 | pub async fn handler(Json(value): Json<Value>) -> impl IntoResponse {
   |                                                   ^^^^^^^^^^^^^^^^^ the trait `IntoResponse` is not implemented for `derive_from_request::Json<serde_json::Value>`
   |
   = help: the following other types implement trait `IntoResponse`:
             &'static [u8; N]
             &'static [u8]
             &'static str
             ()
             (R,)
             (Response<()>, R)
             (Response<()>, T1, R)
             (Response<()>, T1, T2, R)
           and 133 others

For more information about this error, try `rustc --explain E0277`.
```

Is this intentional? It seems the wrapped extractor does not implement `IntoResponse` and can therefore not be used in the same way as `axum::Json`. Should one still use `axum::Json` in this case for the responses in handlers and only the wrapped one as an extractor?

--- a/examples/customize-extractor-error/src/derive_from_request.rs
+++ b/examples/customize-extractor-error/src/derive_from_request.rs
@@ -15,16 +15,25 @@ use axum::{
     extract::rejection::JsonRejection, extract::FromRequest, http::StatusCode,
     response::IntoResponse,
 };
+use serde::Serialize;
 use serde_json::{json, Value};
 
 pub async fn handler(Json(value): Json<Value>) -> impl IntoResponse {
-    Json(dbg!(value));
+    Json(dbg!(value))
 }
 
 // create an extractor that internally uses `axum::Json` but has a custom rejection
 #[derive(FromRequest)]
 #[from_request(via(axum::Json), rejection(ApiError))]
 pub struct Json<T>(T);
+
+// We implement `IntoResponse` for our extractor so it can be used as a response
+impl <T: Serialize> IntoResponse for Json<T> {
+    fn into_response(self) -> axum::response::Response {
+        let Self(value) = self;
+        axum::Json(value).into_response()
+    }
+}
 
 // We create our own rejection type
 #[derive(Debug)]

--- a/examples/customize-extractor-error/src/derive_from_request.rs
+++ b/examples/customize-extractor-error/src/derive_from_request.rs
@@ -28,7 +28,7 @@ pub async fn handler(Json(value): Json<Value>) -> impl IntoResponse {
 pub struct Json<T>(T);
 
 // We implement `IntoResponse` for our extractor so it can be used as a response
-impl <T: Serialize> IntoResponse for Json<T> {
+impl<T: Serialize> IntoResponse for Json<T> {
     fn into_response(self) -> axum::response::Response {
         let Self(value) = self;
         axum::Json(value).into_response()

